File: spring-backend/src/main/java/com/steiner/vblog/model/article/ArticleStatus.java
package com.steiner.vblog.model.article;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

public enum ArticleStatus {
    Draft(0),
    Published(1),
    Dustbin(2);

    public final int code;
    ArticleStatus(int code) {
        this.code = code;
    }

    @JsonValue
    public int getCode() {
        return code;
    }

    @JsonCreator
    public static ArticleStatus fromCode(int code) {
        for (ArticleStatus status : ArticleStatus.values()) {
            if (status.code == code) {
                return status;
            }
        }

        throw new IllegalArgumentException("Invalid status code: %s".formatted(code));
    }
}

File: backend/src/main/kotlin/com/steiner/vblog/model/ArticleStatus.kt
package com.steiner.vblog.model

import com.steiner.vblog.util.ArticleStatusSerializer
import kotlinx.serialization.Serializable

@Serializable(with = ArticleStatusSerializer::class)
enum class ArticleStatus(val code: Int) {
    Draft(0),
    Published(1),
    Dustbin(2),
    Deleted(3)
}
